llvm/lib/Target/SPIRV/SPIRVUtils.cpp

```diff
@@ -525,6 +525,12 @@ size_t PartialOrderingVisitor::GetNodeRank(BasicBlock *BB) const {
       continue;
 
     auto Iterator = BlockToOrder.end();
+    // This block hasn't been ranked yet. Ignoring.
+    // This doesn't happen often, but when dealing with irreducible CFG, we have
+    // to rank nodes without knowing the rank of all their predecessors.
+    if (Iterator == BlockToOrder.end())
```

Comment:
I don't understand how line 527 and 531 work together.

Reply:
They dont't, and I need better test coverage. Somehow I messed up and added this at the wrong place. But somehow all my tests now pass, which means either I lack coverage, or the code below is useless.

Reply:
Thanks for catching my mistake! That code is quite embarrassing to look at 😅

Reply:
All right, extended the test to also verify the rank value (used in the partial traversal function), and fixed the code. Thanks!